src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -75,21 +82,38 @@ withMiningCoordination logger enabled cutDb inner
     | not enabled = inner Nothing
     | otherwise = do
         t <- newTVarIO mempty
-        fmap snd . concurrently (prune t) $ inner . Just $ MiningCoordination
+        c <- newIORef 0
+        fmap snd . concurrently (prune t c) $ inner . Just $ MiningCoordination
             { _coordLogger = logger
             , _coordCutDb = cutDb
             , _coordState = t
-            , _coordLimit = 10000 }
+            , _coordLimit = 2500
+            , _coord503s = c }
   where
-    prune :: TVar MiningState -> IO ()
-    prune t = runForever (logFunction logger) "Chainweb.Chainweb.MinerResources.prune" $ do
-        let !d = 600000000  -- 10 minutes
+    prune :: TVar MiningState -> IORef Int -> IO ()
+    prune t c = runForever (logFunction logger) "Chainweb.Chainweb.MinerResources.prune" $ do
+        let !d = 300000000  -- 5 minutes
         threadDelay d
         ago <- over (_Unwrapped . _Unwrapped) (subtract (int d)) <$> getCurrentTimeIntegral
-        atomically . modifyTVar' t $ over _Unwrapped (M.filter (f ago))
-
-    f :: Time Micros -> T3 a PrevBlock b -> Bool
-    f ago (T3 _ (PrevBlock p) _) = _blockCreationTime p > BlockCreationTime ago
+        m@(MiningState ms) <- atomically $ do
+            ms <- readTVar t
+            modifyTVar' t . over _Unwrapped $ M.filter (f ago)
+            pure ms
+        count <- readIORef c
+        atomicWriteIORef c 0
+        logFunction logger Info . JsonLog $ MiningStats (M.size ms) count (avgTxs m)
+
+    f :: Time Micros -> T3 a PrevTime b -> Bool
+    f ago (T3 _ (PrevTime p) _) = p > BlockCreationTime ago
+
+    avgTxs :: MiningState -> Int
+    avgTxs (MiningState ms) = summed `div` M.size ms
```

Comment:
`div` should be close enough.

Reply:
Would it be better to return the size from before prune? We may be interested in the `max` value.

Reply:
It is already the size from before prune :)